RSSDP/Infrastructure/SsdpCommunicationsServer.cs

```diff
@@ -32,46 +32,39 @@ public sealed class SsdpCommunicationsServer : DisposableManagedObjectBase, ISsd
          * port to use, we will default to 0 which allows the underlying system to auto-assign a free port.
          */
 
-        private object _BroadcastListenSocketSynchroniser = new object();
-        private ISocket _BroadcastListenSocket;
-
-        private object _SendSocketSynchroniser = new object();
-        private List<ISocket> _sendSockets;
-
-        private HttpRequestParser _RequestParser;
-        private HttpResponseParser _ResponseParser;
         private readonly ILogger _logger;
-        private ISocketFactory _SocketFactory;
-        private readonly INetworkManager _networkManager;        
-
-        private int _LocalPort;
-        private int _MulticastTtl;
-
-        private bool _IsShared;
+        private readonly INetworkManager _networkManager;
         private readonly bool _enableMultiSocketBinding;
 
-        /// <summary>
-        /// Raised when a HTTPU request message is received by a socket (unicast or multicast).
-        /// </summary>
-        public event EventHandler<RequestReceivedEventArgs> RequestReceived;
-
-        /// <summary>
-        /// Raised when an HTTPU response message is received by a socket (unicast or multicast).
-        /// </summary>
-        public event EventHandler<ResponseReceivedEventArgs> ResponseReceived;
+        private object _broadcastListenSocketSynchroniser;
+        private ISocket _broadcastListenSocket;
+        private object _sendSocketSynchroniser;
+        private List<ISocket> _sendSockets;
+        private HttpRequestParser _requestParser;
+        private HttpResponseParser _responseParser;
+        private ISocketFactory _socketFactory;
+        private int _localPort;
+        private int _multicastTtl;
+        private bool _isShared;
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="SsdpCommunicationsServer"/> class.
         /// Minimum constructor.
         /// </summary>
         /// <exception cref="ArgumentNullException">The <paramref name="socketFactory"/> argument is null.</exception>
-        public SsdpCommunicationsServer(ISocketFactory socketFactory,
-            INetworkManager networkManager, ILogger logger, bool enableMultiSocketBinding)
+        public SsdpCommunicationsServer(
+            ISocketFactory socketFactory,
+            INetworkManager networkManager,
+            ILogger logger,
+            bool enableMultiSocketBinding)
             : this(socketFactory, 0, SsdpConstants.SsdpDefaultMulticastTimeToLive, networkManager, logger, enableMultiSocketBinding)
         {
-            
+            _broadcastListenSocketSynchroniser = new object();
+            _sendSocketSynchroniser = new object();
```

Comment:
Why?

Reply:
The fields were initialized at declaration and in one constructor. I moved the initialization to the constructors. But both constructors do the same so moving initialization back to declaration.